scripts/sentiment_analysis/text_cnn.py

```diff
@@ -0,0 +1,60 @@
+"""textCNN model."""
+
+import gluonnlp as nlp
+import mxnet as mx
+from mxnet import gluon
+from mxnet.gluon import HybridBlock
+
+class SentimentNet(HybridBlock):
+    """Network for sentiment analysis."""
+
+    def __init__(self, dropout, embed_size=300, vocab_size=100, prefix=None,
+                 params=None, model_mode='multichannel', output_size=2,
+                 num_filters=(100, 100, 100), ngram_filter_sizes=(3, 4, 5)):
+        super(SentimentNet, self).__init__(prefix=prefix, params=params)
+        self.model_mode = model_mode
+        with self.name_scope():
+            self.embedding = gluon.nn.Embedding(vocab_size, embed_size)
+            if self.model_mode == 'multichannel':
+                self.embedding_extend = gluon.nn.Embedding(vocab_size, embed_size)
+                embed_size *= 2
+            self.encoder = nlp.model.ConvolutionalEncoder(embed_size=embed_size,
+                                                          num_filters=num_filters,
+                                                          ngram_filter_sizes=ngram_filter_sizes,
+                                                          conv_layer_activation='relu',
+                                                          num_highway=None)
+            self.output = gluon.nn.HybridSequential()
+            with self.output.name_scope():
+                self.output.add(gluon.nn.Dropout(dropout))
+                self.output.add(gluon.nn.Dense(output_size, flatten=False))
+
+    def hybrid_forward(self, F, data): # pylint: disable=arguments-differ
+        if self.model_mode == 'multichannel':
```

Comment:
good practice i have learnt recently is to kind of mention tensor shapes as comments above the code line in hybrid_forward, this helps is easy readability